singularity-eos/eos/modifiers/zsplit_eos.hpp

```diff
@@ -268,7 +274,16 @@ class ZSplit : public EosBase<ZSplit<ztype, T>> {
  private:
   template <typename Indexer_t = Real *>
   PORTABLE_FORCEINLINE_FUNCTION Real GetIonizationState_(Indexer_t &&lambda) const {
-    return std::max(0.0, lambda[t_.nlambda()]);
+    using namespace variadic_utils;
+    if (is_nullptr(lambda)) {
+      PORTABLE_THROW_OR_ABORT(
+          "StellarCollapse: lambda must contain Ye and 1 space for caching.\n");
+    }
+    if constexpr (is_indexable_v<Indexer_t, IndexableTypes::MeanIonizationState>) {
+      return std::max(0.0, lambda[IndexableTypes::MeanIonizationState()]);
+    } else {
+      return std::max(0.0, lambda[t_.nlambda()]);
```

Comment:
Does this work if I nest modifiers each of which requires lambdas?

Reply:
It does. In fact, that's the idea. With nested modifiers, the lambdas work as a stack---you always treat the end of the lambda array as yours and then work your way inward to the base object.

Reply:
If we had two modifiers (or an EOS and a modifer) that require the same lambda information (I can't think of a good reason right now), the typed indexing would allow them to both use the same information though. With integer indexing, the information would need to be duplicated, yes?

Reply:
:+1: exactly